project2/screens/movie.js

```diff
@@ -1,174 +1,59 @@
-        if (this.state.movieID !== prevState.movieID) {
-            this.lookupMovie(this.state.movieID)
-        }
-    }
-
-    lookupMovie = async (movieID) => {
-        const result = await getMovieDetails(`${movieID}`)
-        console.log(result)
-
-        if (result["Actors"] !== undefined) {
-            this.setState({ actors: result["Actors"] })
-        }
-        if (result["Awards"] !== undefined) {
-            this.setState({ awards: result["Awards"] })
-        }
-        if (result["BoxOffice"] !== undefined) {
-            this.setState({ boxOffice: result["BoxOffice"] })
-        }
-        if (result["Country"] !== undefined) {
-            this.setState({ country: result["Country"] })
-        }
-        if (result["DVD"] !== undefined) {
-            this.setState({ dvd: result["DVD"] })
-        }
-        if (result["Director"] !== undefined) {
-            this.setState({ director: result["Director"] })
-        }
-        if (result["Genre"] !== undefined) {
-            this.setState({ genre: result["Genre"] })
-        }
-        if (result["Language"] !== undefined) {
-            this.setState({ language: result["Language"] })
-        }
-        if (result["Metascore"] !== undefined) {
-            this.setState({ metascore: result["Metascore"] })
-        }
-        if (result["Plot"] !== undefined) {
-            this.setState({ plot: result["Plot"] })
-        }
-        if (result["Poster"] !== undefined) {
-            this.setState({ poster: result["Poster"] })
-        }
-        if (result["Production"] !== undefined) {
-            this.setState({ production: result["Production"] })
-        }
-        if (result["Rated"] !== undefined) {
-            this.setState({ rated: result["Rated"] })
-        }
-        if (result["Released"] !== undefined) {
-            this.setState({ released: result["Released"] })
-        }
-        if (result["Runtime"] !== undefined) {
-            this.setState({ runtime: result["Runtime"] })
-        }
-        if (result["Title"] !== undefined) {
-            this.setState({ title: result["Title"] })
-        }
-        if (result["Type"] !== undefined) {
-            this.setState({ type: result["Type"] })
-        }
-        if (result["Website"] !== undefined) {
-            this.setState({ website: result["Website"] })
-        }
-        if (result["Writer"] !== undefined) {
-            this.setState({ writer: result["Writer"] })
-        }
-        if (result["Year"] !== undefined) {
-            this.setState({ year: result["Year"] })
-        }
-        if (result["imdbID"] !== undefined) {
-            this.setState({ imdbID: result["imdbID"] })
-        }
-        if (result["imdbRating"] !== undefined) {
-            this.setState({ imdbRating: result["imdbRating"] })
-        }
-        if (result["imdbVotes"] !== undefined) {
-            this.setState({ imdbVotes: result["imdbVotes"] })
-        }
-    }
+    lookupMovie();
+  }, []);
```

Comment:
Empty array passed as second argument: `useEffect` fires once when the component mounts, then every time anything defined in this array changes. I only want it to fire when the Movie component mounts (once I've got the data, nothing else needs to be done), so an empty array is passed.